src/lib.rs

```diff
@@ -1,8 +1,84 @@
-#[cfg(test)]
-mod tests {
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
+//! # Dotium
+//!
+//! The uniform object provider for Minecraft Mod hosting platforms.
+//! Implements traits for api wrappers Ferinth, Furse, and Octocrab to then be used platform independedly in GDLauncher
+
+#[cfg(feature = "curseforge")]
+mod curseforge;
+#[cfg(feature = "github")]
+mod github;
+#[cfg(feature = "modrinth")]
+mod modrinth;
+pub mod project;
+pub mod request;
+pub mod version;
+
+use async_trait::async_trait;
+use project::{Author, Project};
+use serde::{Deserialize, Serialize};
+use version::Version;
+
+#[derive(Debug, Clone)]
+pub struct Platform {
+    /// The name of the platform
+    pub name: String,
+    /// The two-letter nick name of the Platform
+    pub short_name: String,
+    /// Logo of the platform
+    pub logo: request::Asset,
+}
+
+#[derive(thiserror::Error, Debug)]
+#[error("The trait is not implemented")]
+pub struct UnimplementedError;
```

Comment:
Wouldn't it be better if it was a trait that depends on std::error:Error?
So that we can fall back to more normal methods if the need arises, or if we need something like the step the error happened for example